src/orion/core/worker/trial.py

```diff
@@ -132,13 +136,15 @@ class Result(Value):
         function or of an 'constraint' expression.
         """
 
+        __slots__ = ('name', '_type', 'value')
```

Comment:
Oh yes. Sorry for that. The tuple defining the slots can be still in `Trial.Value`. What needs to be done is to declare `__slots__ = ()` in `Trial.Param` and `Trial.Result`, so that a namespace `__dict__` won't be created for `Trial.Param` or `Trial.Result`. An object instantiated from the inherited classes will be able only have access to the non-blank namespace defined by the base class.

Reply:
Oh! I will try that.